plugins/kubernetes/test/models/kubernetes/role_validator_test.rb

```diff
@@ -332,10 +332,36 @@
       it "allows static configuration" do
         role.first[:kind] = "ServiceAccount"
         role.first[:spec].delete(:template)
+        role.first[:metadata][:annotations].delete(:"samson/prerequisite")
         refute errors
       end
     end
 
+    describe "#validate_prerequisites_consistency" do
+      before do
+        role.first[:kind] = "Job"
+        role.first[:metadata][:annotations] = {"samson/prerequisite": "true"}
+        role.first[:spec][:template][:spec][:restartPolicy] = "Never"
+
+        role.last[:kind] = "Pod"
```

Comment:
having 2 deployables is kinda weird, best make the second one a ConfigMap or so ... but then that would break the other validation ... so something is not correct here :/

... so logic needs to be:
 - prerequisite needs to be consistent
 - at least one of the prerequisites must be Job/Pod (ideally not anything else deployable, but that's tricky, so let's not do that)

Reply:
also missing test for the `size == 0` condition